internal/repository/repository.go

```diff
@@ -419,7 +419,11 @@ func (r *Repository) saveIndex(ctx context.Context, indexes ...*Index) error {
 
 		debug.Log("Saved index %d as %v", i, sid)
 	}
-	r.idx.MergeFinalIndexes()
+
+	err := r.idx.MergeFinalIndexes()
```

Comment:
`return r.idx.MergeFinalIndexes()` would be simpler

Reply:
Oh, good catch.